src/combinations_with_replacement.rs

```diff
@@ -107,16 +107,17 @@ where
     }
 
     fn nth(&mut self, n: usize) -> Option<Self::Item> {
-        if self.first {
+        let times = if self.first {
             // In empty edge cases, stop iterating immediately
             if !(self.indices.is_empty() || self.pool.get_next()) {
                 return None;
             }
             self.first = false;
-        } else if self.increment_indices() {
-            return None;
-        }
-        for _ in 0..n {
+            n
+        } else {
+            n + 1
```

Comment:
I really do not know how we generally handle this, but `n+1` could overflow.

Reply:
`.nth(usize::MAX)` would overflow.
Previously, `.nth(usize::MAX)` would have allocated `usize::MAX` vectors before returning (so it would take eternity and beyond).
I don't mind reverting back. But I'm not sure all `+1` are safe in our codebase for such exotic use case.

Reply:
I'd go for correctness and revert. On 32-bit systems iterating until usize::MAX might well be reasonable.

Reply:
Done.